routes/api/admin.js

```diff
@@ -48,6 +48,65 @@ router.get('/reservedBooks', function(req,res){
     });
 });
 
+
+//TODO: Clean up with async
+router.post('/getBookInfo', function(req, res, next) {
+
+    //Required parameters
+    var required = ['schoolID','barcode' ,'isbn', 'password'];
+    required.forEach(function(param){
+        if(!req.body[param]){
+            return res.status(400).send();
```

Comment:
Be careful because, if there are more than one missing fields, you will call `send` twice and get an error (not able to send 2 responses in http). The `return` only leaves the current function in the forEach loop.

You can use `some` to stop iterating once you find a missing param:

```
var missingParams = required.some(function(param) {
    return req.body[param] === undefined;
})

if (missingParams) {
    return res.status(400).send();
}
```